.circleci/config.yml

```
@@ -146,3 +145,50 @@ jobs:
                     --attach build/distributions/git-machete-intellij-plugin-*.zip \
                     --message "$tag"$'\n\n'"$change_notes" \
                     $tag
+  checkForNewEapBuildNumber:
+    docker:
+      - image: 'gitmachete/intellij-plugin-ci:6.1.0'
+    working_directory: ~/git-machete-intellij-plugin
+    resource_class: large
+    steps:
+      - checkout
+      - run:
+          name: Reconfigure origin remote to https
+          command: git remote set-url origin https://${GITHUB_TOKEN}@github.com/VirtusLab/git-machete-intellij-plugin.git
+      - run:
+          name: Configure github user information
+          command: |
+            git config user.email "gitmachete@virtuslab.com"
+            git config user.name "Git Machete Bot"
+      - run:
+          # language=sh
+          command: |
+            ./gradlew updateEapBuildNumber
+            if ! git diff --exit-code --quiet intellijVersions.properties ; then
+              newEapBuildNumber=$(grep -Po '(?<=eapOfLatestSupportedMajor=).*' intellijVersions.properties)
+              branch=update-eap/$newEapBuildNumber
+              git checkout -b "$branch"
+              git add .
```

Nit: I avoid `git add .` in such automated scripts as hell. CI pipelines have a general tendency to apply some changes to the project that are not expected in "human" development (and are hence e.g. not protected against via .gitignore).

```suggestion
              git add intellijVersions.properties
```